server/src/main/java/org/opensearch/indices/recovery/PeerRecoveryTargetService.java

```diff
@@ -245,7 +245,7 @@ private void doRecovery(final long recoveryId, final StartRecoveryRequest preExi
                     indexShard.prepareForIndexRecovery();
                     final boolean hasRemoteSegmentStore = indexShard.indexSettings().isRemoteStoreEnabled();
                     if (hasRemoteSegmentStore) {
-                        indexShard.syncSegmentsFromRemoteSegmentStore(false, false, true);
+                        indexShard.syncSegmentsFromRemoteSegmentStore(false, false, false);
```

Comment:
Do we need this commit during a primary relocation?

Reply:
Creating an issue to add testing around primary relocation for when remote store is enabled (#8672). 
Reverting this change, until we can test whether commit is needed during primary relocation.